tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java

```diff
@@ -252,18 +277,32 @@ private static boolean mustBeArray(boolean topValues, String name, RbPath rbPath
             || rbPath.startsWith(RB_METAZONE_INFO);
     }
 
-    private void printArray(RbValue rbValue, boolean quote, boolean isSequence, PrintWriter out) {
-        for (String v : rbValue.getElements()) {
-            newLineAndIndent(out);
-            printValue(out, quoteInside(v), quote);
-            if (!isSequence) {
-                out.print(",");
+    private void printElements(PrintWriter out, RbValue rbValue, FormatOptions format) {
+        // TODO: If "shouldUseComma" is made obsolete, just use the "else" block always.
+        if (rbValue.getElementsPerLine() == 1) {
+            for (String v : rbValue.getElements()) {
+                newLineAndIndent(out, format);
+                printValue(out, quoteInside(v), format);
+                if (format.shouldUseComma) {
+                    out.print(",");
+                }
+            }
+        } else {
+            Iterable<List<String>> partitions =
+                    Iterables.partition(rbValue.getElements(), rbValue.getElementsPerLine());
+            for (List<String> tuple : partitions) {
+                checkArgument(format.shouldUseComma, "cannot group non-sequence values");
```

Comment:
why inside the loop? move this to the top of the else block?

Reply:
Oops, good point.